moveit_ros/visualization/CMakeLists.txt

```diff
@@ -15,103 +22,98 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+find_package(rviz_ogre_vendor REQUIRED)
+find_package(OGRE REQUIRED)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 add_definitions(-DQT_NO_KEYWORDS)
 
-catkin_package(
-  LIBRARIES
-    moveit_motion_planning_rviz_plugin_core
-    moveit_planning_scene_rviz_plugin_core
-    moveit_robot_state_rviz_plugin_core
-    moveit_rviz_plugin_render_tools
-    moveit_trajectory_rviz_plugin_core
-  INCLUDE_DIRS
-    motion_planning_rviz_plugin/include
-    planning_scene_rviz_plugin/include
-    robot_state_rviz_plugin/include
-    rviz_plugin_render_tools/include
-    trajectory_rviz_plugin/include
-  CATKIN_DEPENDS
-    moveit_ros_planning_interface
-    moveit_ros_robot_interaction
-    object_recognition_msgs
-  DEPENDS
-    EIGEN3
+set(THIS_PACKAGE_INCLUDE_DIRS
+  #motion_planning_rviz_plugin/include
+  #planning_scene_rviz_plugin/include
+  #robot_state_rviz_plugin/include
+  rviz_plugin_render_tools/include
+  #trajectory_rviz_plugin/include
+  ${rclcpp_INCLUDE_DIRS}
+  ${Boost_INCLUDE_DIRS}
+  ${EIGEN3_INCLUDE_DIRS}
+  ${QT_INCLUDE_DIR}
 )
 
-catkin_install_python(PROGRAMS scripts/moveit_joy.py DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
-catkin_python_setup()
+set(THIS_PACKAGE_INCLUDE_LIBRARIES
+  #moveit_motion_planning_rviz_plugin_core
+  #moveit_planning_scene_rviz_plugin_core
+  #moveit_robot_state_rviz_plugin_core
+  moveit_rviz_plugin_render_tools
+  #moveit_trajectory_rviz_plugin_core
+)
+
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+  #moveit_ros_planning_interface
+  #moveit_ros_robot_interaction
+  object_recognition_msgs
+  Eigen3
+)
+
+#[[install(
+  PROGRAMS
+  scripts/moveit_joy.py
+  DESTINATION lib/${PROJECT_NAME}
+)]]
 
 include_directories(rviz_plugin_render_tools/include
-                    robot_state_rviz_plugin/include
-                    planning_scene_rviz_plugin/include
-                    motion_planning_rviz_plugin/include
-                    trajectory_rviz_plugin/include
+                    # robot_state_rviz_plugin/include
+                    # planning_scene_rviz_plugin/include
+                    # motion_planning_rviz_plugin/include
+                    # trajectory_rviz_plugin/include
                     ${Boost_INCLUDE_DIRS}
-                    ${catkin_INCLUDE_DIRS})
+                    ${rclcpp_INCLUDE_DIRS})
 
 include_directories(SYSTEM
                     ${EIGEN3_INCLUDE_DIRS}
-                    ${QT_INCLUDE_DIR})
+                    ${QT_INCLUDE_DIR}
+                    ${OGRE_INCLUDE_DIRS})
 
 add_subdirectory(rviz_plugin_render_tools)
-add_subdirectory(robot_state_rviz_plugin)
-add_subdirectory(planning_scene_rviz_plugin)
-add_subdirectory(motion_planning_rviz_plugin)
-add_subdirectory(trajectory_rviz_plugin)
+#add_subdirectory(robot_state_rviz_plugin)
+#add_subdirectory(planning_scene_rviz_plugin)
+#add_subdirectory(motion_planning_rviz_plugin)
+#add_subdirectory(trajectory_rviz_plugin)
 
 install(FILES
-  motion_planning_rviz_plugin_description.xml
```

Comment:
you should replace this line with 
```
pluginlib_export_plugin_description_file(rviz_common plugins_description.xml)
```

Have a look here https://github.com/AcutronicRobotics/rviz_hrim_plugins/blob/master/CMakeLists.txt

Reply:
Used this for the robot_state_rviz_plugin_description, will have to do the same with others once ported. 
https://github.com/nzlz/moveit2/blob/3eeba3abb3d9a5c8c3e61ff0c02308260de50e5c/moveit_ros/visualization/CMakeLists.txt#L107
`pluginlib_export_plugin_description_file(rviz_common robot_state_rviz_plugin_description.xml)`